pali/COPYING

```diff
@@ -0,0 +1,19 @@
+voc.txt and output.txt are derived from the Digital Pāḷi Dictionary which is licensed under a Creative Commons License: https://digitalpalidictionary.github.io/
+
+To create the files, I downloaded their latest SQL Database and ran the following query:
+
+```sql
+SELECT DISTINCT lemma_2, stem
+FROM dpd_headwords
+WHERE
+  stem IS NOT "*" AND
+  stem IS NOT "-" AND 
+  stem NOT LIKE "!%";
+```
+
+I then exported the lemma_2 column to voc.txt and the stem column to output.txt.
```

Comment:
Your `output.txt` needs to contain what your Snowball pali stemmer actually produces when fed your `voc.txt` - currently the tests fail due to this not being the case.

The stems produce don't necessarily need to be the linguistically correct.  What actually matters is that forms of the same word get mapped to the same stem, and (especially important) that words which aren't forms of the same word get mapped to different stems.

The stems are often the actual root word just because that's often a natural way to work but for example the English stemmer stems `trying` to `tri` (along with `try`, `tries`, `tried`, `trier`, etc) even though the linguistic root would be `try`.

Also some understemming is unlikely to be a problem (the aim is to improve search, and a stemmer which understems some cases will almost inevitably beat not stemming at all).  Overstemming is more likely problematic as it results in false matches.

Reply:
I noticed there are quite a few duplicate words in `pali/voc.txt`:

```
pali/voc.txt: Ignoring duplicate of word 'anati'
pali/voc.txt: Ignoring duplicate of word 'asati'
pali/voc.txt: Ignoring duplicate of word 'asanta'
pali/voc.txt: Ignoring duplicate of word 'assati'
pali/voc.txt: Ignoring duplicate of word 'āhari'
pali/voc.txt: Ignoring duplicate of word 'upaṭṭhāsi'
pali/voc.txt: Ignoring duplicate of word 'oraka'
pali/voc.txt: Ignoring duplicate of word 'kassati'
pali/voc.txt: Ignoring duplicate of word 'tāvataka'
pali/voc.txt: Ignoring duplicate of word 'dassati'
pali/voc.txt: Ignoring duplicate of word 'nessati'
pali/voc.txt: Ignoring duplicate of word 'apariggaha'
pali/voc.txt: Ignoring duplicate of word 'pāti'
pali/voc.txt: Ignoring duplicate of word 'pāpaka'
pali/voc.txt: Ignoring duplicate of word 'pāyāsi'
pali/voc.txt: Ignoring duplicate of word 'bhāti'
pali/voc.txt: Ignoring duplicate of word 'bhoti'
pali/voc.txt: Ignoring duplicate of word 'sañjāti'
pali/voc.txt: Ignoring duplicate of word 'samādapenta'
pali/voc.txt: Ignoring duplicate of word 'samāna'
pali/voc.txt: Ignoring duplicate of word 'samūhanati'
pali/voc.txt: Ignoring duplicate of word 'sammati'
pali/voc.txt: Ignoring duplicate of word 'saranta'
pali/voc.txt: Ignoring duplicate of word 'āsurasakkarakkhasā'
pali/voc.txt: Ignoring duplicate of word 'vihanati'
pali/voc.txt: Ignoring duplicate of word 'akukkuccaka'
pali/voc.txt: Ignoring duplicate of word 'aññāti'
pali/voc.txt: Ignoring duplicate of word 'asanta'
pali/voc.txt: Ignoring duplicate of word 'appamāṇaka'
pali/voc.txt: Ignoring duplicate of word 'pakāsaka'
pali/voc.txt: Ignoring duplicate of word 'vañcako'
pali/voc.txt: Ignoring duplicate of word 'bhassanta'
pali/voc.txt: Ignoring duplicate of word 'pasupati'
```

and also some words which start `-` (maybe these are valid but they seem likely to be artifacts of the data source):

```
$ grep '-' pali/voc.txt 
-anti
-iṃ
-iṃsu
-uṃ
-eyyo
-tara
-ti
-tu
-tho
-mo
-mi
-si
-hi
```

Reply:
Thanks @ojwb for taking a look. Yeah, my intention was to craft a good set of test cases first (this is not good yet! but might get there), then write a stemmer that does pretty well (95%?) against those tests, then filter out the handful of cases that it can't get right before merging.